scripts/sources/parameters/cmd/aws_ssm.go

```diff
@@ -0,0 +1,58 @@
+/*
+Based on: https://github.com/awsdocs/aws-doc-sdk-examples/blob/main/gov2/ssm/GetParameter/GetParameterv2.go
+*/
+package parameters
+
+import (
+	"context"
+	"log"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/ssm"
+)
+
+// SSMGetParametersAPI and SSMDescribeParametersAPI defines the interface
+// for the GetParameters and DescribeParameters function.
+// We use this interface to test the function using a mocked service
+type SSMGetParametersAPI interface {
+	GetParameters(
+		ctx context.Context,
+		params *ssm.GetParametersInput,
+		optFns ...func(*ssm.Options),
+	) (*ssm.GetParametersOutput, error)
+}
+
+type SSMDescribeParametersAPI interface {
+	DescribeParameters(
+		ctx context.Context,
+		params *ssm.DescribeParametersInput,
+		optFns ...func(*ssm.Options),
+	) (*ssm.DescribeParametersOutput, error)
+}
+
+// ExecGetParameters and ExecDescribeParameters retrieves an AWS Systems Manager string parameter
+// Inputs:
+// 		c: is the context of the method call, which includes the AWS Region
+// 		api: is the interface that defines the method call
+// 		input: defines the input arguments to the service call
+// Output:
+// 		If success, a GetParametersOutput object containing the result of the service call and nil
+// 		Otherwise, nil and an error from the call to GetParameters
+func ExecGetParameters(c context.Context, api SSMGetParametersAPI, input *ssm.GetParametersInput) (*ssm.GetParametersOutput, error) {
```

Comment:
This seems like too many layers of indirection. Why not just instantiate a struct with that interface and make the call. What do you gain by having this extra function in the middle?